src/handwritten/_install_daqmx.py

```diff
@@ -0,0 +1,185 @@
+import urllib.request
+import json
+import tempfile
+import platform
+import os
+import subprocess
+import winreg
+import ctypes
+import sys
+
+# <TODO> repelace the URL with a static URL
+_META_DATA_URL = "http://localhost:8000/ni-daqmx-update-metadata.json"
+
+
+def _is_admin():
+    """
+    Verify whether the user has Administrator privileges.
+    """
+    try:
+        return ctypes.windll.shell32.IsUserAnAdmin()
+    except Exception as e:
+        print(f"Error checking admin privileges: {e}")
+        return False
+
+
+def _parse_version(version):
+    """
+    Split the version string into an array of integers.
+    """
+    return [int(part) for part in version.split(".")]
+
+
+def _compare_version(installed_ver, source_ver):
+    """
+    Compare two versions which are in the format: major.minor.update
+    """
+    iver_parts = _parse_version(installed_ver)
+    sver_parts = _parse_version(source_ver)
+
+    for iver_num, sver_num in zip(iver_parts + [0], sver_parts + [0]):
+        if iver_num < sver_num:
+            return -1
+        elif iver_num > sver_num:
+            return 1
```

Comment:
Convert the versions to tuples and their comparison operators will automatically do lexical comparisons. The tuples don't have to have the same number of elements.

This is often used with [sys.version_info](https://docs.python.org/3/library/sys.html#sys.version_info):
- `if sys.version_info >= (3, 9):`
- `if sys.version_info < (3, 12, 2):`